packages/opentelemetry-core/src/context/propagation/B3Propagator.ts

```diff
@@ -38,25 +48,106 @@ function isValidSpanId(spanId: string): boolean {
   return VALID_SPANID_REGEX.test(spanId) && !INVALID_ID_REGEX.test(spanId);
 }
 
+function isValidParentSpanID(spanId: string | undefined): boolean {
+  return spanId === undefined || isValidSpanId(spanId);
+}
+
+function isValidSampledValue(sampled: number | undefined): boolean {
+  return (
+    sampled === undefined ||
+    VALID_SAMPLED_VALUES.includes(sampled) ||
+    VALID_UNSAMPLED_VALUES.includes(sampled)
+  );
+}
+
+function parseHeader(header: unknown) {
+  return Array.isArray(header) ? header[0] : header;
+}
+
 /**
  * Propagator for the B3 HTTP header format.
  * Based on: https://github.com/openzipkin/b3-propagation
  */
 export class B3Propagator implements HttpTextPropagator {
+  _getHeaderValue(carrier: unknown, getter: GetterFunction, key: string) {
```

Comment:
I assume these underscore prefixed methods are meant to be `private`?

Reply:
yeah

Reply:
I will remove them. It was part of another review comment, but I don't like them.

Reply:
I also don't love them. Function overhead is nontrivial